keras_core/ops/image.py

```diff
@@ -102,3 +102,133 @@ def resize(
     return backend.image.resize(
         image, size, method=method, antialias=antialias, data_format=data_format
     )
+
+
+class AffineTransform(Operation):
+    def __init__(
+        self,
+        method="bilinear",
+        fill_mode="constant",
+        fill_value=0,
+        data_format="channels_last",
+    ):
+        super().__init__()
+        self.method = method
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.data_format = data_format
+
+    def call(self, image, transform):
+        return backend.image.affine_transform(
+            image,
+            transform,
+            method=self.method,
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            data_format=self.data_format,
+        )
+
+    def compute_output_spec(self, image, transform):
+        if len(image.shape) not in (3, 4):
+            raise ValueError(
+                "Invalid image rank: expected rank 3 (single image) "
+                "or rank 4 (batch of images). Received input with shape: "
+                f"image.shape={image.shape}"
+            )
+        if len(transform.shape) not in (1, 2):
+            raise ValueError(
+                "Invalid transform rank: expected rank 1 (single transform) "
+                "or rank 2 (batch of transforms). Received input with shape: "
+                f"transform.shape={transform.shape}"
+            )
+        return KerasTensor(image.shape, dtype=image.dtype)
+
+
+@keras_core_export("keras_core.ops.image.affine_transform")
+def affine_transform(
+    image,
+    transform,
+    method="bilinear",
+    fill_mode="constant",
+    fill_value=0,
+    data_format="channels_last",
+):
+    """Applies the given transform(s) to the image(s).
+
+    Args:
+        image: Input image or batch of images. Must be 3D or 4D.
+        transform: Projective transform matrix/matrices. A vector of length 8 or
+            tensor of size N x 8. If one row of transform is
+            `[a0, a1, a2, b0, b1, b2, c0, c1]`, then it maps the output point
+            `(x, y)` to a transformed input point
+            `(x', y') = ((a0 x + a1 y + a2) / k, (b0 x + b1 y + b2) / k)`,
+            where `k = c0 x + c1 y + 1`. The transform is inverted compared to
+            the transform mapping input points to output points. Note that
+            gradients are not backpropagated into transformation parameters.
+            Note that `c0` and `c1` are only effective when using TensorFlow
+            backend and will be considered as `0` when using other backends.
+        method: Interpolation method. Available methods are `"nearest"`,
```

Comment:
I just noticed that we have a slight API inconsistency: this function as well as `resize` both use `method`, but our preprocessing layers use `interpolation`. Perhaps we should just switch everything (including `resize`) to `interpolation` since the preprocessing layers are much older and more widely used.

Reply:
I have replaced `method` with `interpolation` in both `affine_transform` and `resize`.
Additionally, there have been some API changes in keras-core.